Hi Thomas, thanks a lot for the PR and sorry it took me a while to to look at it.
I've reproduced the issue and I agree that we should check for initdb early.

I get a compiler warning after applying your commit though:
```
pg_checksums.c:485:8: warning: unused variable ‘ret’ [-Wunused-variable]
  int   ret;
        ^~~
```

Also, but I guess that is unrelated to your PR, I wonder whether we should clarify the error message a bit, e.g. by saying `pg_checksums: program "initdb" not found in path` and/or by mentioning that it is required in order to sync `PGDATA` (which is probably non-obvious to the end-user), do you have a good suggestion for rewording?

--- a/pg_checksums.c
+++ b/pg_checksums.c
@@ -441,7 +441,6 @@ static void findInitDB(const char *argv0)
 static void
 syncDataDir(char *DataDir)
 {
-	int			ret;
 	char		cmd[MAXCMDLEN];
 
 

--- a/pg_checksums.c
+++ b/pg_checksums.c
@@ -601,7 +601,8 @@ main(int argc, char *argv[])
 	if (activate || verify)
 #if PG_VERSION_NUM < 100000
 		// check for initdb
-		findInitDB(argv[0]);
+		if (activate)
+			findInitDB(argv[0]);
 #endif
 	{
 		/* Scan all files */

--- a/pg_checksums.c
+++ b/pg_checksums.c
@@ -398,7 +398,7 @@ updateControlFile(char *DataDir, ControlFileData *ControlFile)
 }
 
 /*
- * syncDataDir() is used on PostgreSQL releases <= 10, only
+ * syncDataDir() and findInitDB are used on PostgreSQL releases < 10, only
  */
 #if PG_VERSION_NUM < 100000
 

